sw/host/opentitanlib/src/image/image.rs

```diff
@@ -120,6 +147,41 @@ impl Image {
         Ok(())
     }
 
+    pub fn get_sigverify_params_from_manifest(&self) -> Result<SigverifyParams> {
+        let manifest = self.borrow_manifest()?;
+        let manifest_def: ManifestSpec = manifest.try_into()?;
+
+        let pub_key = manifest_def
+            .pub_key()
+            .ok_or(ImageError::Parse)?
+            .to_le_bytes();
+
+        let signature = manifest_def
+            .signature()
+            .ok_or(ImageError::Parse)?
+            .to_le_bytes();
+
+        if (manifest.manifest_version.major == CHIP_MANIFEST_VERSION_MAJOR1)
+            && (manifest.manifest_version.minor == CHIP_MANIFEST_VERSION_MINOR1)
+        {
+            let rsa_key = RsaPublicKey::new(Modulus::from_le_bytes(pub_key)?)?;
+            let rsa_sig = RsaSignature::from_le_bytes(signature)?;
+            return Ok(SigverifyParams::Rsa(rsa_key, rsa_sig));
+        }
+
+        let ecdsa_pub_key = EcdsaRawPublicKey {
+            x: pub_key[0..32].try_into()?,
+            y: pub_key[32..64].try_into()?,
+        };
```

Comment:
There is also the "reader" API of `EcdsaRaw{PublicKey,Signature}`:

```
let ecdsa_pub_key = EcdsaRawPublicKey::read(std::io::Cursor::new(pub_key))?;
let ecdsa_sig = EcdsaRawSignature::read(std::io::Cursor::new(signature))?;
```

Reply:
Fixed.